main.go

```diff
@@ -21,38 +21,34 @@ func download_manager(hrefs chan []string, parser chan *goquery.Document){
   }
 }
 
-func downloader(url string, page chan *goquery.Document) {
+func downloader(url string, page chan<- *goquery.Document) {
   doc, _ := goquery.NewDocument(url)
-  //if err != nil {
-    //log.Fatal(err)
-  //}
-
-  page <- doc
+  if doc != nil {
+    page <- doc
+  }
 }
 
-//
 // Parse the specified __page__ for all links
 //  Send the number of links on that page to be analyzed to printer
 //  Send the links on the page to be downloaded
-func parse(page chan *goquery.Document, printer chan *page_data, hrefs chan []string){
+func parse(page <-chan *goquery.Document, printer chan<- *page_data, hrefs chan<- []string){
   links := make([]string, 0)
-  select {
-  case doc := <-page:
+  for doc := range page {
     doc.Find("a").Each(func(i int, s *goquery.Selection) {
       link, _ := s.Attr("href")
       links = append(links, link)
     })
 
-    hrefs <- links
-
-    result := &page_data{doc.Url.String(), len(links)}
-    printer <- result
+    go func() {
```

Comment:
Read about unbuffered and buffered channels. Buffered channels will block only when they're full. Unbuffered channels are literally synchronous. The write blocks until it's read.